optimism/src/mips/witness.rs

```diff
@@ -813,7 +813,12 @@ impl<Fp: Field> Env<Fp> {
 
     fn snapshot_state_at(&mut self, at: &StepFrequency) {
         if self.should_trigger_at(at) {
-            let filename = format!("snapshot-state-{}.json", self.instruction_counter);
+            let snapshot_directory =
+                std::env::var("ZKVM_SNAPSHOT_DIRECTORY").unwrap_or(".".to_string());
```

Comment:
There is a command-line argument for that as well (see https://github.com/o1-labs/proof-systems/pull/1523#discussion_r1425375789)

Reply:
Well not really for the directory parameter.

Reply:
I totally agree, it is far better with the parameter. I don't see how to use it tbh. Why was it added?

Reply:
It's a Cannon Cli argument, so we need to be able to parse it 😆